codex/codex.nim

```diff
@@ -133,7 +133,15 @@ proc new*(T: type CodexServer, config: CodexConf): T =
 
     wallet = WalletRef.new(EthPrivateKey.random())
     network = BlockExcNetwork.new(switch)
-    localStore = SQLiteStore.new(config.dataDir / "repo", cache = cache)
+    repoDir = config.dataDir / "repo"
+
+  if io2.createPath(repoDir).isErr:
+    trace "Unable to create data directory for block store", dataDir = repoDir
+    raise (ref Defect)(
+      msg: "Unable to create data directory for block store: " & repoDir)
+
+  let
+    localStore = SQLiteStore.new(repoDir, cache = cache)
```

Comment:
Since `SQLiteStore` (nor `SQLiteDatastore`) creates the directory where the SQLite database will be stored, it's necessary for the executable to do this.

It wasn't necessary to do this previously because `FSStore` implicitly handles directory creation
https://github.com/status-im/nim-codex/blob/a7fdb975df56b5be1e6f724389b182e31e7ccfc3/codex/stores/fsstore.nim#L89-L93

We could modify `FSStore.new` to check if `repoDir` exists and fail if it does not.

Reply:
It's OK for now, but we'll probably move this to our repo abstraction once we have one.